src/pages/projectSettings/CreateDuplicateProjectButton.tsx

```diff
@@ -0,0 +1,103 @@
+import { cloneElement, useState } from "react";
+import { useQuery } from "@apollo/client";
+import Button, { Size, Variant } from "@leafygreen-ui/button";
+import Icon from "@leafygreen-ui/icon";
+import { Menu, MenuItem } from "@leafygreen-ui/menu";
+import { zIndex } from "constants/tokens";
+import { GetUserPermissionsQuery } from "gql/generated/types";
+import { GET_USER_PERMISSIONS } from "gql/queries";
+import { CopyProjectModal } from "./CopyProjectModal";
+import { CreateProjectModal } from "./CreateProjectModal";
+import { ProjectType } from "./tabs/utils";
+
+const NewProjectButton = (
+  <Button
+    size={Size.Small}
+    data-cy="new-project-button"
+    leftGlyph={<Icon glyph="Plus" />}
+    variant={Variant.Primary}
+  >
+    New Project
+  </Button>
+);
+
+interface Props {
+  id: string;
+  label: string;
+  owner: string;
+  projectType: ProjectType;
+  repo: string;
+}
+
+export const CreateDuplicateProjectButton: React.VFC<Props> = ({
+  id,
+  label,
+  owner,
+  projectType,
+  repo,
+}) => {
+  const { data } = useQuery<GetUserPermissionsQuery>(GET_USER_PERMISSIONS);
+  const {
+    user: {
```

Comment:
what is represented by the object after the colon?

Reply:
It's a list of permissions that the current user possesses. We can add more fields in the future to conditionally render features in Spruce
https://github.com/evergreen-ci/evergreen/blob/bd8cf591918180b803cf0232cac9146b87599a31/graphql/schema.graphql#L1129-L1132